webapi/Controllers/ChatController.cs

```diff
@@ -179,8 +176,7 @@ public class ChatController : ControllerBase, IDisposable
 
         // Verify that the chat exists and that the user has access to it.
         ChatSession? chat = null;
-#pragma warning disable CA1508 // Avoid dead conditional code. It's giving out false positives on chat == null.
-        if (!(await chatSessionRepository.TryFindByIdAsync(chatId, callback: c => chat = c)) || chat == null)
```

Comment:
The old code seemed to imply we could get a ChatSession that would be null. Is that possible?  If so, we are no longer covering that case...

Reply:
Yeah, I saw that too.  The nullability in the definition is due to how the TryFindByIdAsync is doing the assignment in a callback.  Let's just call it an antipattern that wasn't introduced here (out of scope).